### PR TITLE
Build profile docs change related to sleep

### PR DESCRIPTION
**Pull request by bulislaw**
This change will be needed when https://github.com/ARMmbed/mbed-os/pull/4097 is merged.

### CODE REVIEW

**Comment by bulislaw** [comment]
@AnotherButler

File: docs/dev_tools/build_profiles.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ mbed OS 5 supports three primary build profiles: *develop*, *debug* and *release
 * Largest and slowest profile.
 * Full error information. For example, asserts have file name and line number.
 * Easy to step through code with a debugger.
+* Sleep is disabled.
```
Comment (AnotherButler) [inline docs/dev_tools/build_profiles.md]: To eliminate passive voice, I recommend changing this to "Disabled sleep" or "Disabled sleep mode". Would that keep the same meaning?

**Comment by bulislaw** [comment]
Good point, will update.

**Comment by Patater** [comment]
There is probably a way to rejigger things so you can avoid repeating yourself.

**Comment by bulislaw** [comment]
I think it's cleaner to repeat small amount of text than to make some 'common' options with exclusions.

**Comment by Patater** [comment]
Yes, a repeating small amount is much more readable.

However, the issues listed here for when sleep is enabled sound like good candidates for a known issues document. This location is probably not the first place a user would expect to find these known issues, although it may be handy to link to them from here. It's also probably difficult to keep these caveats up to date as they aren't as visible as a known issues document which would get reviewed on every release.

**Comment by bulislaw** [comment]
I agree, unfortunately our docs are not necessarily very well structured or clear to find things in. https://docs.mbed.com/docs/mbed-os-handbook/en/latest/
To be frank I wouldn't be able easily find known issues (I think they are in blog posts per release but not in docs themselves).

**Comment by Patater** [comment]
As an outsider, I'd look for known issues in the GitHub issue tracker or in a dedicated release notes document.

Anyway, making sure these issues are in the GitHub issue tracker or in release notes is not really something for this PR to deal with.

File: docs/dev_tools/build_profiles.md [hunk 2]
```diff
@@ -5,11 +5,15 @@ mbed OS 5 supports three primary build profiles: *develop*, *debug* and *release
 * Small and fast code.
 * Full error information. For example, asserts have file name and line number.
 * Hard to follow code flow when using a debugger.
+* Chip is put to sleep when going idle:
```
Comment (AnotherButler) [inline docs/dev_tools/build_profiles.md]: Query: Could we change this to "Chip goes to sleep we idle" and keep the meaning? I would like to avoid the passive voice in "is put to sleep".

Comment (Patater) [inline docs/dev_tools/build_profiles.md]: "The RTOS idle task puts the chip to sleep." is accurate, but maybe too detailed.

"The chip sleeps when idle" is accurate as well.

**Comment by AnotherButler** [comment]
@geky @theotherjimmy Should I merge this? The instructions say to merge it after we merge [ARMmbed/mbed-os#4097](https://github.com/ARMmbed/mbed-os/pull/4097), but I see [ARMmbed/mbed-os#4200](https://github.com/ARMmbed/mbed-os/pull/4200) reverts that.

**Comment by geky** [comment]
I would wait, it does look like the revert will go through, but I suspect a slightly revised pr will be merged later.

**Comment by AnotherButler** [comment]
@geky Thanks. I'll do that.

**Comment by AnotherButler** [comment]
@geky Should I merge this yet or still wait?

**Comment by bulislaw** [comment]
The revert PR was closed https://github.com/ARMmbed/mbed-os/pull/4200 I think that should be merged to reflect what's going on in the code. Then we should update the docs again when/if it changes.
